pkg/inframetadata/internal/hostmap/updater.go

```diff
@@ -0,0 +1,89 @@
+package hostmap
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/multierr"
+)
+
+// updater handles updates to a hostmetadata payload,
+// recording changes and (non-fatal) errors.
+type updater struct {
+	// changed records if there have been any changes.
+	changed bool
+	// attrs stores attributes to use for update of the payload.
+	attrs pcommon.Map
+	// err records any non-fatal errors.
+	err error
+}
+
+// newUpdater creates a new Updater.
```

Comment:
What is `res` used for?

Reply:
We read its contents for updates; the `updater` handles extracting values from a resource and checking their types

Reply:
not using the updater anymore, marking this as solved